packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,230 @@
+import json
+from typing import Any, Dict, List, Optional, Set
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models.hackathon_teams_models import (
+    HackathonTeam,
+    MoveTeamMembers,
+    TeamType,
+    UpdateTeam,
+)
+from py_api.utilities.parsers import filter_none_values
+from pymongo.results import UpdateResult
+
+
+class TeamsController:
+    @classmethod
+    def create_team(
+            cls, team_name: str, user_id: str = "",
+            members_list: Optional[List[str]] = None,
+    ) -> HackathonTeam | None:
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+            team_type = TeamType.RANDOM
+            team_name = cls.generate_random_team_name()
+
+        team_members = []
+
+        if user_id:
+            team_members = [user_id]
+
+        if members_list:
+            team_members = list(members_list)
+            # Assumes that we are creating a random team when moving members from existing team to a non-existing one
+            team_type = TeamType.RANDOM
+
+        if cls.fetch_team(team_name):
+            # A name with this team_name exist so no new team is created
+            return None
+
+        new_team = HackathonTeam(
+            team_name=team_name,
+            team_type=team_type, team_members=team_members,
+        )
+        t_col.insert_one(new_team.model_dump())
+
+        return new_team
+
+    @classmethod
+    def add_team_participant(cls, team_name: str, user_id: str) -> List[str] | None:
```

Comment:
Why would you expect the method add_team_participant to return a list of members instead of the object?

Reply:
Call this one update_team_members and make the method accept either one object or a list. "Overload" the method but in the pythonic way.